### PR TITLE
WeatherApp - Production

### PR DESCRIPTION
**Pull request by davidolanrewaju**
## 🌦 WeatherApp - Production

>This PR details what was done to build the project

### Implemented Features ♻
- [x] Installed all required dependencies
- [x] Made API calls
- [x] Created a city list as the homepage
- [x] Change view when any of the cities is clicked
- [x] Display weather of clicked city in the new view

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,185 @@
...
+- Twitter: [@ola_dayve](https://twitter.com/ola_dayve)
+- LinkedIn: [David Olanrewaju](https://www.linkedin.com/in/david-olanrewaju)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- VIDEO SUMMARY -->
+## 🎥 Video Summary <a name="video-summary"></a>
+>This is a video summary on the aim of this project and the challenges encountered.</br>
+Click [this link](https://www.loom.com/share/f02739603fca4d7c83ee79ed7d6c995a) to get a view of the project
+
```
Comment (TadesseDev) [inline README.md:135]: - [ ] Dear @davidolanrewaju, You have don great on the Implementation, I also, I like your video presentation. But, Please remember the video is expected to contain a highlight about a piece of the code that you find interesting or you are proud of making. Please, consider redoing the video with the `piece of the code that you find interesting` alongside the objective description and demo of the project.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,185 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank microverse for the resources provided to make this project possible.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (TadesseDev) [inline README.md:162]: - [ ] Also, here. The README file is expected to contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).  Please add the appropriate Acknowledgements for [Nelson Sakwa](https://www.behance.net/sakwadesignstudio)

File: src/tests/Cities.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <Provider store={store}>
+      <MemoryRouter>
+        <Cities />
+      </MemoryRouter>
+    </Provider>
+  );
+
+  expect(getByText("City 1")).toBeInTheDocument();
+  expect(getByText("City 2")).toBeInTheDocument();
+});
```
Comment (TadesseDev) [inline src/tests/Cities.test.js:39]: - [ ] Dear @davidolanrewaju, Good job writing the tests here. But I figure you only have this and the other two snapshot tests. Please remember tests are a great tool to keep our functionalities in place and assert everything is working well. Please add some tests for the pure functions in the Redux code;
- [Optional] I can see you have made use of react testing library, But I would love to suggest adding more tests. This will give you the freedom of changing things in the future considering your test will fail if you break any functionalities.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @davidolanrewaju 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Good job overall following the template design ✅
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Cities.test.js
Comment (KanzaTahreem) [inline src/tests/Cities.test.js:1]: - [ ] Good job with the tests 🥇🥇 However as suggested [here](https://github.com/davidolanrewaju/WeatherApp/pull/5#discussion_r1224739443), kindly add few more tests for the pure functions in the Redux code🤓⛏ For example, in `Cities.test.js`, since you have mocked `useNavigate ` you can add tests to ensure that clicking on a city navigates to the correct route✨ 
- Similarly, for the `DisplayForecast.test.js`, you can test an `error` state and add more assertions for other relevant data displayed. Another example could be testing an empty `list: [ ]`
- Likewise, in `DisplayWeather.test.js` you can test `getWeather` function. Tests can be written to mock the API response and ensure that the function handles the response correctly 💪

  _Please note that this is just an idea. These are not tested, you can add you own tests. However, please make sure to add some more tests as suggested earlier 😃_`

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @davidolanrewaju 🙋‍♀️

Good job so far implementing a few of the suggested changes, I really appreciate the effort you put in!  **_However, there are few tings that still need to be foxed But you are almost there!_** 💪 🏁

so_close

## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Great design
- ✅ Descriptive PR and README file
- ✅ Highlighted a piece of code


## Required Changes 🛑


_Please check the comments under the review._

## Optional suggestions ❕
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Cities.test.js
Comment (davidolanrewaju) [reply src/tests/Cities.test.js:1]: That was an optional suggestion ma'am @KanzaTahreem

Comment (KanzaTahreem) [reply src/tests/Cities.test.js:1]: Hello @davidolanrewaju, every comment with the optional prefix is optional. Please look at the previous reviewers comment. The first comment does not have an optional prefix 😄 It is with the second one.
_If you think that the review was unfair, you can ask for a second opinion using the form given above_ ✨
 I wish you success 💪


IMG_20230610_110710

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @davidolanrewaju, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👏 You did well adding more tests as requested by previous reviewers.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
